File: Linkpoint/src/main/java/com/linkpoint/slproto/SLCircuitInfo.java
package com.linkpoint.slproto;

import com.linkpoint.slproto.auth.SLAuthReply;
import java.net.InetSocketAddress;
import java.net.SocketAddress;
import java.util.UUID;

public class SLCircuitInfo {
    public final UUID agentID;
    final int circuitCode;
    public final UUID sessionID;
    final SocketAddress socketAddress;

    SLCircuitInfo(SLAuthReply sLAuthReply) {
        this.socketAddress = new InetSocketAddress(sLAuthReply.simAddress, sLAuthReply.simPort);
        this.sessionID = sLAuthReply.sessionID;
        this.agentID = sLAuthReply.agentID;
        this.circuitCode = sLAuthReply.circuitCode;
    }
}

File: Linkpoint/src/main/java/com/linkpoint/slproto/SLCircuitInfo.kt
package com.linkpoint.slproto

import com.linkpoint.slproto.auth.SLAuthReply
import java.net.InetSocketAddress
import java.net.SocketAddress
import java.util.UUID

class SLCircuitInfo internal constructor(authReply: SLAuthReply) {
    val socketAddress: SocketAddress = InetSocketAddress(authReply.simAddress, authReply.simPort)
    val sessionID: UUID = authReply.sessionID
    val agentID: UUID = authReply.agentID
    val circuitCode: Int = authReply.circuitCode
}